### DISCUSSION TITLE
How do I change the web cert on Security Onion 2?

### DISCUSSION THREAD

**Discussion by chromefinch** [Security-Onion-Solutions/securityonion#1758] [Unsupported Versions] [answered]
How do I change the web cert on Security Onion 2? 
I knew how to do it on the Ubuntu/apache version but am not having much luck with the new hotness.

**Comment by TOoSmOotH** [comment, marked as answer]
I created #1766 for this. Right now if you replace it, it will overwrite your cert with a freshly generated one. You can modify https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/nginx/init.sls#L78-L79 under default and replace managerssl.crt/key with something you drop in /etc/pki on the manager. Just don't name it managerssl. Keep in mind the salt code will get overwritten next upgrade so you would need to do it again until we fix it but we will see if we can sneak this into the next release.

**Reply by chromefinch** [reply]
oh wow! thank you a bunch!! 
That seems doable, and thanks for adding it to the feature request!

**Reply by Xboarder56** [reply]
@TOoSmOotH after replacing the managerssl.crt/key I noticed osquery would fail with a cert generation for the old self signed cert. What is your recommendation to handle the additional services trying to use the old self signed cert and not working anymore? I was thinking of replacing ca.crt/key but not sure if it could cause further issues.
